do not use ambiguous "end"
do not compare input file stream with integer
author: Samuel Moors (Free University of Brussels)
--- ./Chrysalis/GraphFromFasta_MPI.cc.orig	2017-11-21 15:10:35.915874040 +0100
+++ ./Chrysalis/GraphFromFasta_MPI.cc	2017-11-21 15:11:16.751868585 +0100
@@ -49,7 +49,7 @@
 
 
 #include<sys/time.h>
-static struct timeval start,end;
+static struct timeval start,stop;
 void timer_start(){
 
   gettimeofday(&start,NULL);
@@ -57,8 +57,8 @@
 
 
 double timer_stop(){
-  gettimeofday(&end,NULL);
-  double time_taken = ((end.tv_usec-start.tv_usec) + 1000000*(end.tv_sec - start.tv_sec));
+  gettimeofday(&stop,NULL);
+  double time_taken = ((stop.tv_usec-start.tv_usec) + 1000000*(stop.tv_sec - start.tv_sec));
  time_taken = time_taken/1000;
   return time_taken;
 }
--- ./Chrysalis/ReadsToTranscripts_MPI_chang.cc.orig	2017-10-20 14:42:21.000000000 +0200
+++ ./Chrysalis/ReadsToTranscripts_MPI_chang.cc	2017-11-21 15:43:02.489700214 +0100
@@ -34,7 +34,7 @@
 }
 
 #include<sys/time.h>
-static struct timeval start,end;
+static struct timeval start,stop;
 void timer_start(){
 
   gettimeofday(&start,NULL);
@@ -42,8 +42,8 @@
 
 
 double timer_stop(){
-  gettimeofday(&end,NULL);
-  double time_taken = ((end.tv_usec-start.tv_usec) + 1000000*(end.tv_sec - start.tv_sec));
+  gettimeofday(&stop,NULL);
+  double time_taken = ((stop.tv_usec-start.tv_usec) + 1000000*(stop.tv_sec - start.tv_sec));
  time_taken = time_taken/1000;
   return time_taken;
 }
--- ./Chrysalis/ReadsToTranscripts_MPI.cc.orig	2017-10-20 14:42:21.000000000 +0200
+++ ./Chrysalis/ReadsToTranscripts_MPI.cc	2017-11-21 15:40:23.293798438 +0100
@@ -38,7 +38,7 @@
 
 #include<sys/time.h>                                                                                                                                                                                                                      
     
-static struct timeval start,end;                                                                                                                                                                                                           
+static struct timeval start,stop;                                                                                                                                                                                                           
    
 void timer_start(){                                                                                                                                                                                                                        
    
@@ -53,8 +53,8 @@
                        
 double timer_stop(){                                                                                                                                                                                            
                               
-    gettimeofday(&end,NULL);                                                                                                                                                                                                               
-    double time_taken = ((end.tv_usec-start.tv_usec) + 1000000*(end.tv_sec - start.tv_sec));                                                                                                                                               
+    gettimeofday(&stop,NULL);                                                                                                                                                                                                               
+    double time_taken = ((stop.tv_usec-start.tv_usec) + 1000000*(stop.tv_sec - start.tv_sec));                                                                                                                                               
     time_taken = time_taken/1000;                                                                                                                                                                                                         
     return time_taken;                                                                                                                                                                                                                     
      
--- ./Chrysalis/analysis/sequenceUtil.cc.orig	2017-11-21 14:34:37.449436229 +0100
+++ ./Chrysalis/analysis/sequenceUtil.cc	2017-11-21 14:36:07.653477550 +0100
@@ -50,7 +50,7 @@
 string read_sequence_from_file (string filename) {
   
   ifstream fileReader (filename.c_str());
-  if (fileReader == 0) { // couldn't open file
+  if (!fileReader) { // couldn't open file
 	throw(stacktrace() + "\n\nCould not open " + filename + "\n");
   }
   
